resources/clj-kondo.exports/lilactown/helix/clj_kondo/lilactown/helix.clj

```diff
@@ -0,0 +1,69 @@
+(ns clj-kondo.lilactown.helix
+  (:require [clj-kondo.hooks-api :as api]))
+
+#_{:clj-kondo/ignore [:clojure-lsp/unused-public-var]}
+(defn $
+  "Macro analysis for `helix.core/$`."
+  [{:keys [node]}]
+  (let [[dollar-sym component-sym & body] (-> node :children)
+        old-props                         (-> body first :children)
+        children                          (-> body rest)
+        new-props                         (->> old-props
+                                               (map #(if (cond-> (api/sexpr %) symbol? (= '&))
+                                                       (api/keyword-node :&)
+                                                       %)))
+        expanded                          (api/list-node (list*
+                                                          dollar-sym
+                                                          component-sym
+                                                          (api/map-node new-props)
+                                                          children))]
+    (comment
+      ;; FIXME: @lilactown I'd like your opinion on this diagnostic and its
+      ;; semantics before committing.
+      (when (->> children
+                 (some api/map-node?))
+        (api/reg-finding! (doto (merge {:message "Map passed as children"
+                                        :type    :helix/invalid-children}
+                                       (meta children))))))
```

Comment:
_technically_ `children` can be anything, depending on what the component that's being rendered does with it. i could see us adding additional checks for like `helix.dom/$d` et al. for checking children, but i wouldn't want to restrict valid types being passed as children

Reply:
Want me to then delete this rhc?

Reply:
yeah then i think i'm good to merge